test/Base_Tests/src/Data/Decimal_Spec.enso

```diff
@@ -643,6 +643,51 @@ add_specs suite_builder =
+            Decimal.new "34.56" . to_text . should_equal "34.56"
+            Decimal.new "34.56" . to_display_text . should_equal "34.56"
+            Decimal.new "123E80" . to_text . should_equal "1.23E+82"
+            Decimal.new "123E80" . to_display_text . should_equal "1.23E+82"
+            Decimal.new "0.00016E-20" . to_text . should_equal "1.6E-24"
+            Decimal.new "0.00016E-20" . to_display_text . should_equal "1.6E-24"
+
+            Decimal.new "-34.56" . to_text . should_equal "-34.56"
+            Decimal.new "-34.56" . to_display_text . should_equal "-34.56"
+            Decimal.new "-123E80" . to_text . should_equal "-1.23E+82"
+            Decimal.new "-123E80" . to_display_text . should_equal "-1.23E+82"
+            Decimal.new "-0.00016E-20" . to_text . should_equal "-1.6E-24"
+            Decimal.new "-0.00016E-20" . to_display_text . should_equal "-1.6E-24"
+
+        group_builder.specify "should convert to text without scientific notation correctly" <|
+            Decimal.new "34.56" . to_text_without_scientific_notation . should_equal "34.56"
+            Decimal.new "123E80" . to_text_without_scientific_notation . should_equal "12300000000000000000000000000000000000000000000000000000000000000000000000000000000"
+            Decimal.new "0.00016E-20" . to_text_without_scientific_notation . should_equal "0.0000000000000000000000016"
+
+            Decimal.new "-34.56" . to_text_without_scientific_notation . should_equal "-34.56"
+            Decimal.new "-123E80" . to_text_without_scientific_notation . should_equal "-12300000000000000000000000000000000000000000000000000000000000000000000000000000000"
+            Decimal.new "-0.00016E-20" . to_text_without_scientific_notation . should_equal "-0.0000000000000000000000016"
+
+        group_builder.specify "should format correctly" <|
+            d = Decimal.new "123456789.87654"
+
+            d.format . should_equal "123,456,789.87654"
+            d.format locale=Locale.italy . should_equal "123.456.789,87654"
+            d.format format="#,###.##" . should_equal "123,456,789.88"
+            d.format format="#,###.##" locale=Locale.italy . should_equal "123.456.789,88"
+            d.format format="#,##.##" . should_equal "1,23,45,67,89.88"
+            d.format format="#,##.##" locale=Locale.italy . should_equal "1.23.45.67.89,88"
+
+        group_builder.specify "should parse correctly" <|
+            Decimal.parse "123456789.87654" . should_equal 123456789.87654
+            Decimal.parse "123,456,789.87654" locale=Locale.default . should_equal 123456789.87654
+            Decimal.parse "123,456,789.87654" locale=Locale.us . should_equal 123456789.87654
+            Decimal.parse "123.456.789,87654" locale=Locale.italy . should_equal 123456789.87654
+            Decimal.parse "123,456,789.87654" locale=Locale.italy . should_equal 123.456
+            Decimal.parse "123.456.789,87654" locale=Locale.us . should_equal 123.456
```

Comment:
I'm wondering if we should allow setting the custom format in `parse` as well.

What if I have a string like `1.23.45.67.89,88` (that I got from somewhere) and want to parse it?

But that can be done in future iterations.

Reply:
https://github.com/enso-org/enso/issues/9674